pkg/providers/amifamily/bottlerocket.go

```diff
@@ -88,9 +88,97 @@ func (b Bottlerocket) UserData(kubeletConfig *v1.KubeletConfiguration, taints []
 			CustomUserData:      customUserData,
 			InstanceStorePolicy: instanceStorePolicy,
 		},
+		AMIVersion:       b.resolveAMIVersion(),
+		AMISelectorTerms: b.AMISelectorTerms,
 	}
 }
 
+// resolveAMIVersion extracts AMI version from selector terms
+func (b Bottlerocket) resolveAMIVersion() string {
+	if version := b.getVersionFromAlias(); version != "" {
+		return version
+	}
+	if version := b.getVersionFromName(); version != "" {
+		return version
+	}
+	if version := b.getVersionFromID(); version != "" {
+		return version
+	}
+	if version := b.getVersionFromResolvedAMIs(); version != "" {
+		return version
+	}
+	return ""
+}
+
+func (b Bottlerocket) getVersionFromAlias() string {
+	for _, term := range b.AMISelectorTerms {
+		if term.Alias == "" {
+			continue
+		}
+		parts := strings.Split(term.Alias, "@")
+		if len(parts) == 2 {
+			return parts[1]
+		}
+	}
+	return ""
+}
+
+func (b Bottlerocket) getVersionFromName() string {
+	for _, term := range b.AMISelectorTerms {
+		if term.Name == "" {
+			continue
+		}
+		if version := extractVersionFromName(term.Name); version != "" {
+			return version
+		}
+	}
+	return ""
+}
+
+func (b Bottlerocket) getVersionFromID() string {
+	for _, term := range b.AMISelectorTerms {
+		if term.ID == "" {
+			continue
+		}
+		if version := b.findVersionByID(term.ID); version != "" {
+			return version
+		}
+	}
+	return ""
+}
+
+func (b Bottlerocket) findVersionByID(id string) string {
+	for _, ami := range b.AMIs {
+		if ami.ID == id && ami.Name != "" && strings.Contains(ami.Name, "bottlerocket") {
+			return extractVersionFromName(ami.Name)
+		}
+	}
+	return ""
+}
+
+func (b Bottlerocket) getVersionFromResolvedAMIs() string {
+	for _, ami := range b.AMIs {
+		if ami.Name == "" || !strings.Contains(ami.Name, "bottlerocket") {
+			continue
+		}
+		if version := extractVersionFromName(ami.Name); version != "" {
+			return version
+		}
+	}
+	return ""
+}
+
+func extractVersionFromName(name string) string {
```

Comment:
nit: I'd scope this to the Bottlerocket object still just so id doesn't pollute the package namespace

```suggestion
func (Bottlerocket) extractVersionFromName(name string) string {
```